File: 02 Binary/Missing Number/MissingNum.java
import java.util.Scanner;

class Solution {
    public int missingNumber(int[] nums) {
        int xor = 0;
        for (int i = 0; i < nums.length; i++) {
            xor = xor ^ (i + 1) ^ nums[i];
        }
        return xor;
    }
}

public class MissingNum {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            int n = sc.nextInt();
            int[] nums = new int[n];
            for (int i = 0; i < n; i++) {
                nums[i] = sc.nextInt();
            }
            Solution obj = new Solution();
            System.out.println(obj.missingNumber(nums));
            t--;
        }
        sc.close();
    }
}

File: 02 Binary/Missing Number/MissingNum.go
package main

import "fmt"

func missingNumber(nums []int) int {
	xor := 0
	for i, num := range nums {
		xor = xor ^ (i + 1) ^ num
	}
	return xor
}

func main() {
	var t int
	fmt.Scan(&t)
	for t > 0 {
		t--
		var n int
		fmt.Scan(&n)
		nums := make([]int, n)
		for i := 0; i < n; i++ {
			fmt.Scan(&nums[i])
		}
		fmt.Println(missingNumber(nums))
	}
}
